pkg/daemon/criruntime/imageruntime/helpers.go

```diff
@@ -230,31 +230,11 @@ func (r *imagePullStatusReader) mainloop() {
 
 func (c ImageInfo) ContainsImage(name string, tag string) bool {
 	for _, repoTag := range c.RepoTags {
-		imageRepo, imageTag := parseRepositoryTag(repoTag)
+		// Ref: https://github.com/openkruise/kruise/issues/1273
+		imageRepo, imageTag, _ := daemonutil.NormalizeImageRefToNameTag(repoTag)
```

Comment:
For example, we want to pre-download the image `whitematrixlab/astar-wasm:astar-wasm-v0.0.2`. 
1,The name of the downloaded image through CRI.PullImage is like this `docker.io/whitematrixlab/astar-wasm:astar-wasm-v0.0.2`.
2,  As the following code, before downloading image, it will be checked whether it already exists:

```go

func (w *pullWorker) getImageInfo(ctx context.Context) (*runtimeimage.ImageInfo, error) {
	imageInfos, err := w.runtime.ListImages(ctx)   
	if err != nil {
		klog.V(5).Infof("List images failed, err %v", err)
		return nil, err
	}
	for _, info := range imageInfos {
		if info.ContainsImage(w.name, w.tagSpec.Tag) {
			return &info, nil
		}
	}
	return nil, fmt.Errorf("image %v:%v not found", w.name, w.tagSpec.Tag)
}
```
The image name in the `imageInfos`  is  `docker.io/whitematrixlab/astar-wasm:astar-wasm-v0.0.2`. The value of the `w.name` variable is `whitematrixlab/astar-wasm:astar-wasm-v0.0.2`. 

3,  The `defaultDomain` and `officialRepoName`  are not processed in the `parseRepositoryTag` method. As the following code, it will be considered that the image does not exist
```go
func (c ImageInfo) ContainsImage(name string, tag string) bool {
	for _, repoTag := range c.RepoTags {
		imageRepo, imageTag := parseRepositoryTag(repoTag)
		if imageRepo == name && imageTag == tag {  // it will never be true, (docker.io/whitematrixlab/astar-wasm==whitematrixlab/astar-wasm,astar-wasm-v0.0.2==astar-wasm-v0.0.2)
			return true
		}
	}
	return false
}
```

Due to the above reasons, it will lead to repeated execution of the download logic(It returns results quickly, because the processing of `CRI.PullImage` will check whether the image already exists.)

Reply:
Good, can you add some UT function, and contains the above case.